src/imagings/requests/NewImagingRequest.tsx

```diff
@@ -94,19 +114,56 @@ const NewImagingRequest = () => {
         <Alert color="danger" title={t('states.error')} message={t(error.message || '')} />
       )}
       <form>
-        <div className="form-group patient-typeahead">
-          <Label htmlFor="patientTypeahead" isRequired text={t('imagings.imaging.patient')} />
-          <Typeahead
-            id="patientTypeahead"
-            placeholder={t('imagings.imaging.patient')}
-            onChange={(p: Patient[]) => onPatientChange(p[0])}
-            onSearch={async (query: string) => PatientRepository.search(query)}
-            searchAccessor="fullName"
-            renderMenuItemChildren={(p: Patient) => <div>{`${p.fullName} (${p.code})`}</div>}
-            isInvalid={!!error.patient}
-            feedback={t(error.patient as string)}
-          />
-        </div>
+        <Row>
+          <Column>
+            <div className="form-group patient-typeahead">
+              <Label htmlFor="patientTypeahead" isRequired text={t('imagings.imaging.patient')} />
+              <Typeahead
+                id="patientTypeahead"
+                placeholder={t('imagings.imaging.patient')}
+                onChange={(p: Patient[]) => {
+                  onPatientChange(p[0])
+                }}
+                onSearch={async (query: string) => PatientRepository.search(query)}
+                searchAccessor="fullName"
+                renderMenuItemChildren={(p: Patient) => <div>{`${p.fullName} (${p.code})`}</div>}
+                isInvalid={!!error.patient}
+                feedback={t(error.patient as string)}
+              />
+            </div>
+          </Column>
+          <Column>
+            <div className="visits">
+              {newImagingRequest.patient && visitOption ? (
+                <SelectWithLabelFormGroup
+                  name="visit"
+                  label={t('imagings.imaging.visit')}
+                  isRequired
+                  isEditable
+                  options={visitOption}
+                  defaultSelected={visitOption.filter(
+                    ({ value }) => value === newImagingRequest.visitId,
+                  )}
+                  onChange={(values) => {
+                    onVisitChange(values[0])
+                  }}
+                />
+              ) : (
+                <SelectWithLabelFormGroup
+                  name="visit"
+                  label={t('imagings.imaging.visit')}
+                  isRequired
+                  isEditable={false}
+                  options={[]}
+                  onChange={(values) => {
+                    onVisitChange(values[0])
+                  }}
```

Comment:
since `defaultSelected` takes an undefined, I think we can simplify this logic to the following (or something similar to this): 

```
<SelectWithLabelFormGroup
  name="visit"
  label={t('imagings.imaging.visit')}
  isRequired
  isEditable={patient !== undefined}
  options={patient.visits?.map((v) => ({ label: v.type, value: v.id })) || []}
  onChange={(values) => {
 onVisitChange(values[0])
}}
          />
```